packages/svelte-urql/src/queryStore.ts

```diff
@@ -0,0 +1,85 @@
+import { pipe, concatMap, subscribe, filter } from 'wonka';
+import type { OperationContext } from '@urql/core';
+import { derived, writable } from 'svelte/store';
+import { createRequest } from '@urql/core';
+import type {
+  AnnotatedOperationResult,
+  UrqlStore,
+  Pausable,
+  UrqlStoreArgs,
+} from './common';
+import {
+  defaultBaseResult,
+  createPausable,
+  fetchProcess,
+  fromStore,
+} from './common';
+
+/**
+ * Creates a Svelte store for an [Urql query](https://formidable.com/open-source/urql/docs/api/core/#clientexecutequery) using [Wonka](https://wonka.kitten.sh/)
+ */
+export function queryStore<Data, Variables extends object = {}>(
+  args: UrqlStoreArgs<Data, Variables> & {
+    /** initial value for `isPaused$` return (default is `false`) */
+    pause?: boolean;
+  }
+) {
+  // create the graphql request
+  const request = createRequest(args.query, args.variables);
+
+  // `args.context.requestPolicy` beats `args.requestPolcy`
+  const context: Partial<OperationContext> = {
+    requestPolicy: args.requestPolicy,
+    ...args.context,
+  };
+
+  // combine default with operation details
+  const baseResult: AnnotatedOperationResult<Data, Variables> = {
+    ...defaultBaseResult,
+    operation: args.client.createRequestOperation('query', request),
+  };
+
+  // create a store for fetch results
+  const writableResult$ = writable(baseResult);
+
+  // create a store for `Pausable` interface (defaults to false)
+  // package es2015 doesn't support nullish coalescing operator (??)
+  const isPaused$ = writable(args.pause ? true : false);
+
+  // record when the fetch is complete
+  let isComplete = false;
+
+  // make the store reactive (ex: change when we receive a response)
+  const wonkaSubscription = pipe(
+    // have wonka subscribe to the pauseStore
+    fromStore(isPaused$),
+
+    // don't continue if paused
+    filter(isPaused => !isPaused),
+
+    // now we want to fetch the return type, so we must concatMap
+    concatMap(() =>
+      fetchProcess(
+        args.client.executeQuery<Data, Variables>(request, context),
+        baseResult,
+        () => (isComplete = true)
```

Comment:
This unfortunately isn't possible 😅 
If it was then we'd be happy to just let everyone use promises in Svelte. However, the reason why we expose sources and not promises or promise-likes is that a result is never complete. A query can always receive updates. Basically, be it with a normalised cache or without, a query may receive multiple updates while it's active, i.e. while it has subscribers ([There's more details on the architecture page](https://formidable.com/open-source/urql/docs/architecture/#requests-and-operations-on-the-client))

This kind of answered another question, I had. At least, I think it does 😅 Which was: How do ensure there's no memory leak when we never unsubscribe (which is what I saw in the `subscriptionStore` file)
The unsubscribe functionality never seems to be passed through from stores to internal subscriptions :v:

Reply:
> How do ensure there's no memory leak when we never unsubscribe

Thank you for this great point about memory leak.  i will update to ensure unsubscribe is called when the store has no more subscribers.

> A query can always receive updates

Another good point: even though graphql fetch will only return a single result, `executeQuery` could deliver a series of changes (from cache/fetch/exchange)? I will fix this.

i can set each to unsubscribe when the store is no longer in use (also to ensure no memory leak).

Reply:
fixed in https://github.com/FormidableLabs/urql/pull/2370/commits/e2f837b3d98115e1b6f8932732eff23bd40ea322